platform/src/main/java/org/hillview/sketches/Centroids.java

```diff
@@ -0,0 +1,101 @@
+package org.hillview.sketches;
+
+import org.hillview.dataset.api.IJson;
+import org.hillview.table.api.IMembershipSet;
+import org.hillview.table.api.ITable;
+import org.hillview.utils.BlasConversions;
+import org.jblas.DoubleMatrix;
+import org.jblas.ranges.AllRange;
+import org.jblas.ranges.PointRange;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * This class represents a set of centroids of a partitioning of a table. The number of centroids should not be too
+ * large, as they are shipped over the network.
+ */
+public class Centroids implements IJson {
+    /**
+     * Encodes the position of the centroids. Every row is a centroid, and the columns are the dimensions.
+     * Each partition has a centroid, which is simply the mean of the nD points in the partition.
+     * Should be computed with computeCentroids() by the head node.
+     */
+    @Nullable
+    public DoubleMatrix centroids;
+    /**
+     * Holds the sum of the values in the partitions. Every row is the sum of all rows belonging to the corresponding
+     * partition.
+     */
+    public DoubleMatrix sums;
```

Comment:
This requires you to have a numbering of the clusters into dense integers (the rows of this matrix). I think that it would be better to have a cluster be described by a key of some type T, and the sums should be a HashMap<T, DoubleVector>. Then the centroids of categories can use the category string for the keys, so it will be a Centroids<String> subclass.